app/commands/exercise/approach/link_matching_submissions.rb

```diff
@@ -0,0 +1,45 @@
+class Exercise::Approach::LinkMatchingSubmissions
+  include Mandate
+
+  initialize_with :approach
+
+  def call
+    if approach.tags.empty?
+      relink_submissions!(linked_submissions)
```

Comment:
Sounds good. You could change this to be a guard.

```suggestion
    return relink_submissions!(linked_submissions) if approach.tags.empty?
```